runtime/metricsview/executor_summary.go

```diff
@@ -0,0 +1,302 @@
+	defaultTimeDimName := e.metricsView.TimeDimension
+
+	for _, dim := range e.metricsView.Dimensions {
+		// Skip dimensions that the user cannot access first
+		if !e.security.CanAccessField(dim.Name) {
+			continue
+		}
+
+		// If we have reached the limit, stop adding dimensions
+		if len(dimensions)+len(timeDimensions) >= DefaultSummaryLimit {
+			break
+		}
+
+		// Check if this dimension is a time dimension (has timestamp data type)
+		isTimeDimension := false
+		if dim.DataType != nil {
+			switch dim.DataType.Code {
+			case runtimev1.Type_CODE_TIMESTAMP, runtimev1.Type_CODE_DATE, runtimev1.Type_CODE_TIME:
+				isTimeDimension = true
+			}
+		}
+
+		if isTimeDimension {
+			timeDimensions = append(timeDimensions, dim)
+		} else {
+			dimensions = append(dimensions, dim)
+		}
+	}
+
+	// Ensure the default time dimension is included even if it's not in the dimensions list
+	defaultTimeDimFound := false
+	if defaultTimeDimName != "" {
+		for _, dim := range timeDimensions {
+			if dim.Name == defaultTimeDimName {
+				defaultTimeDimFound = true
+				break
+			}
+		}
+	}
+
+	// Get the default time range for the metrics view using the default time dimension
+	// This will be used for time range filtering in other queries
+	var defaultTimeRange TimestampsResult
+	var err error
+	if defaultTimeDimName != "" {
+		defaultTimeRange, err = e.Timestamps(ctx, defaultTimeDimName)
+		if err != nil {
+			return nil, fmt.Errorf("failed to get default time range for dimension %q: %w", defaultTimeDimName, err)
+		}
+	}
+
+	timeDimensionSummaries := make([]DimensionSummary, 0, len(timeDimensions)+1)
+
+	// Process each time dimension found in the dimensions list
+	for _, dim := range timeDimensions {
+		timeRange, err := e.Timestamps(ctx, dim.Name)
+		if err != nil {
+			return nil, fmt.Errorf("failed to get time range for dimension %s: %w", dim.Name, err)
+		}
+
+		var dataType string
+		if dim.DataType != nil {
+			dataType = dim.DataType.Code.String()
+		}
+
+		summary := DimensionSummary{
+			Name:     dim.Name,
+			DataType: dataType,
+		}
+
+		// Only populate min/max if we have time data
+		if !timeRange.Min.IsZero() {
+			summary.MinValue = timeRange.Min
+		}
+		if !timeRange.Max.IsZero() {
+			summary.MaxValue = timeRange.Max
+		}
+
+		timeDimensionSummaries = append(timeDimensionSummaries, summary)
+	}
+
+	// Include default time dimension if it wasn't found in explicit dimensions
+	if !defaultTimeDimFound && defaultTimeDimName != "" && e.security.CanAccessField(defaultTimeDimName) {
+		summary := DimensionSummary{
+			Name:     defaultTimeDimName,
+			DataType: "TIMESTAMP", // Assume timestamp for default time dimension
+		}
+
+		// Only populate min/max if we have time data
+		if !defaultTimeRange.Min.IsZero() {
+			summary.MinValue = defaultTimeRange.Min
+		}
+		if !defaultTimeRange.Max.IsZero() {
+			summary.MaxValue = defaultTimeRange.Max
+		}
+
+		timeDimensionSummaries = append(timeDimensionSummaries, summary)
+	}
```

Comment:
Since this relates to the logic that builds `defaultTimeDimFound` on lines 77-85, I suggest moving this block up right below that code.